File: benchmark/benchmark.py
#!/usr/bin/env python3

import pyasn
import time

addrs_path = "addrs.txt"
db_path = "ipasn_20190822.dat"

def read_addrs(path):
    return list(map(str.strip, open(path)))

if __name__ == '__main__':
    print('Loading database from {}'.format(db_path))
    start = time.time_ns()
    asndb = pyasn.pyasn(db_path)
    print('Took {}ms\n'.format((time.time_ns()-start)/1e6))

    addrs = read_addrs(addrs_path)

    print('Looking up {} addresses (string)'.format(len(addrs)))
    start = time.time_ns()
    for addr in addrs:
        asndb.lookup(addr)
    print('Took {}ms'.format((time.time_ns()-start)/1e6))

File: benchmark/benchmark.go
package main

import (
	"bufio"
	"fmt"
	"net"
	"os"
	"time"

	"github.com/maxmouchet/goasn/goasn"
)

var addrsPath = "addrs.txt"
var dbPath = "ipasn_20190822.dat"

func readAddrsAsStr(path string) []string {
	file, _ := os.Open(path)
	defer file.Close()

	addrs := make([]string, 0)
	scanner := bufio.NewScanner(file)

	for scanner.Scan() {
		addrs = append(addrs, scanner.Text())
	}

	return addrs
}

func readAddrsAsIP(path string) []net.IP {
	addrsStr := readAddrsAsStr(path)
	addrsIP := make([]net.IP, len(addrsStr))
	for i, addr := range addrsStr {
		addrsIP[i] = net.ParseIP(addr)
	}
	return addrsIP
}

func main() {
	fmt.Printf("Loading database from %s...\n", dbPath)
	start := time.Now()
	asndb, _ := goasn.NewDB(dbPath)
	fmt.Printf("Took %s\n\n", time.Since(start))

	addrsStr := readAddrsAsStr(addrsPath)
	addrsIP := readAddrsAsIP(addrsPath)

	fmt.Printf("Looking up %d addresses (string)\n", len(addrsStr))
	start = time.Now()
	for _, addr := range addrsStr {
		asndb.LookupStr(addr)
	}
	fmt.Printf("Took %s\n\n", time.Since(start))

	fmt.Printf("Looking up %d addresses (net.IP)\n", len(addrsIP))
	start = time.Now()
	for _, addr := range addrsIP {
		asndb.LookupIP(addr)
	}
	fmt.Printf("Took %s\n", time.Since(start))
}
